data_license.go

```diff
@@ -0,0 +1,233 @@
+package thingfulx
+
+// Category is a simple type alias for a string
+
+const (
+
+	// Permissions type
+
+	// Reproduction is "making multiple copies"
+	Reproduction = "https://creativecommons.org/ns#Reproduction"
+
+	// Distribution is "distribution, public display, and publicly performance"
+	Distribution = "https://creativecommons.org/ns#Distribution"
+
+	// DerivativeWorks is "distribution of derivative works"
+	DerivativeWorks = "https://creativecommons.org/ns#DerivativeWorks"
+
+	// Sharing is "permits commercial derivatives, but only non-commercial distribution"
+	Sharing = "https://creativecommons.org/ns#Sharing"
+
+	// Requirements type
+
+	// Notice is "copyright and license notices be kept intact"
+	Notice = "https://creativecommons.org/ns#Notice"
+
+	// Attribution is "credit be given to copyright holder and/or author"
+	Attribution = "https://creativecommons.org/ns#Attribution"
+
+	// ShareAlike is "derivative works be licensed under the same terms or compatible terms as the original work"
+	ShareAlike = "https://creativecommons.org/ns#ShareAlike"
+
+	// SourceCode is "source code (the preferred form for making modifications) must be provided when exercising some rights granted by the license"
+	SourceCode = "https://creativecommons.org/ns#SourceCode"
+
+	// Copyleft is "derivative and combined works must be licensed under specified terms, similar to those on the original work"
+	Copyleft = "https://creativecommons.org/ns#Copyleft"
+
+	// LesserCopyleft is "derivative works must be licensed under specified terms, with at least the same conditions as the original work; combinations with the work may be licensed under different terms"
+	LesserCopyleft = "https://creativecommons.org/ns#LesserCopyleft"
+
+	// Prohibitions type
+
+	// CommercialUse is "exercising rights for commercial purposes"
+	CommercialUse = "http://web.resource.org/cc/CommercialUse"
+
+	// HighIncomeNationUse is "use in a non-developing country"
+	HighIncomeNationUse = "https://creativecommons.org/ns#HighIncomeNationUse"
+)
+
+// DataLicense is struct used to describe data license
+type DataLicense struct {
+	// Human readable name for this license
+	Name string `json:"name"`
+
+	// url to the short version of the license
+	URL string `json:"id"`
+
+	// url to the long or legal version of the license
+	LegalCodeURL string `json:"legalCodeUrl"`
```

Comment:
i really think just `legalCode` is ok for the JSON key here - do you feel strongly about this @dotmancando?